src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,16 +1,205 @@
-import { Component } from 'react';
-
-type Props = {};
-type State = {};
-
-export class NewMovie extends Component<Props, State> {
-  state: State = {};
-
-  render() {
-    return (
-      <form>
-        Put the form here
-      </form>
-    );
-  }
-}
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Movie } from '../../react-app-env';
+import './NewMovie.scss';
+
+type Props = {
+  addMovie: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ addMovie }) => {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [hasTitle, setTitleError] = useState(false);
+  const [hasImgUrl, setImgUrlError] = useState(false);
+  const [hasImdbUrl, setImdbUrlError] = useState(false);
+  const [hasImdbId, setImdbIdError] = useState(false);
```

Comment:
let's name stateful variables properly, setter and value should have corresponding names
```suggestion
  const [titleError, setTitleError] = useState(false);
  const [imgUrlError, setImgUrlError] = useState(false);
  const [imdbUrlError, setImdbUrlError] = useState(false);
  const [imdbIdError, setImdbIdError] = useState(false);
```